### PR TITLE
[Bug] fixed the ignore restored option bug

### LINKED ISSUE #1838: [Bug] ignore restored option is not valid when starting the job from savepoint
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Selecting the ignore restored option does not work when the program is restored from savepoint but the job logic is modified

### StreamPark Version

1.2.4-snapshot

### Java Version

1.8

### Flink Version

1.13.6/1.14.6

### Scala Version of Flink

2.11/2.12

### Error Exception

```log
Failed to rollback to checkpoint/savepoint hdfs:///user/flink/job-ck/test_mult_insert/savepoint-0099dc-f829fdf835d0. Cannot map checkpoint/savepoint state for operator 9dd63673dd41ea021b896d5203f3ba7c to the new program, because the operator is not available in the new program. If you want to allow to skip this, you can set the --allowNonRestoredState option on the CLI.
```

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by youyangkou**
## What changes were proposed in this pull request

Issue Number: close  [1838](https://github.com/apache/incubator-streampark/issues/1838)

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by 1996fanrui** [comment]
Hi @youyangkou , thanks for your contribution. Do you run job in standalone or yarn?

I see the `submitRequest.savepointRestoreSettings` has been set to the PackagedProgram. I will analyze why it doesn't work tomorrow.

```
    val packageProgram = PackagedProgram
      .newBuilder
      .setSavepointRestoreSettings(submitRequest.savepointRestoreSettings)
      .setJarFile(jarFile)
      .setEntryPointClassName(flinkConfig.getOptional(ApplicationConfiguration.APPLICATION_MAIN_CLASS).get())
      .setArguments(
        flinkConfig
          .getOptional(ApplicationConfiguration.APPLICATION_ARGS)
          .orElse(Lists.newArrayList()): _*
      ).build()
```

**Comment by 1996fanrui** [comment]
Do you run your job in the application mode? 

After read related code, I'm sure the `savepointRestoreSettings` doesn't take effect in the yarn or k8s application mode.

**Comment by youyangkou** [comment]
> Yes, I tested the application and session modes of k8s and yarn and this parameter does not work

**Comment by 1996fanrui** [comment]
>  Yes, I tested the application and session modes of k8s and yarn and this parameter does not work

Do you mean it doesn't take affect in the k8s session mode?

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala [hunk 1]
```diff
@@ -105,6 +105,9 @@ trait FlinkSubmitTrait extends Logger {
     val retainedOption = CheckpointingOptions.MAX_RETAINED_CHECKPOINTS
     flinkConfig.set(retainedOption, flinkDefaultConfiguration.get(retainedOption))
 
+    //set savepoint.ignore-unclaimed-state parameter
+    flinkConfig.setBoolean(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE, submitRequest.savepointRestoreSettings.allowNonRestoredState())
```
Comment (1996fanrui) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: As I understand, `SavepointConfigOptions.SAVEPOINT_PATH` and `SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE` can replace the `submitRequest.savepointRestoreSettings`. And the `submitRequest.savepointRestoreSettings` isn't called in application mode, so it's a bug.

So I suggest remove the `submitRequest.savepointRestoreSettings`.

Comment (1996fanrui) [reply streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: Hi @wolfboys , please help double check this solution in your free time, thanks~

Comment (wolfboys) [reply streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: > Hi @wolfboys , please help double check this solution in your free time, thanks~

👌 I'll check it leater

**Review by wolfboys — COMMENTED** [review]
hi youyang:

Thank you for your contribution! the bug does exist, and your current commit fixed it, but the code can be further improved:

1) delete 163 lines of code in FlinkSubmitTrait:
`.setSavepointRestoreSettings(submitRequest.savepointRestoreSettings)`

2) `savepointRestoreSettings` need  improvement

before:
```
  lazy val savepointRestoreSettings: SavepointRestoreSettings = {
    lazy val allowNonRestoredState = Try(extraParameter.get(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE.key).toString.toBoolean).getOrElse(false)
    savePoint match {
      case sp if Try(sp.isEmpty).getOrElse(true) => SavepointRestoreSettings.none
      case sp => SavepointRestoreSettings.forPath(sp, allowNonRestoredState)
    }
  }
```
after:
```
  lazy val allowNonRestoredState = Try(extraParameter.get(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE.key).toString.toBoolean).getOrElse(false)

```
3) add code line 101 in FlinkSubmitTrait:

 ```      
.safeSet(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE, submitRequest.allowNonRestoredState)
```

**Comment by youyangkou** [comment]
> hi youyang:
> 
> Thank you for your contribution! the bug does exist, and your current commit fixed it, but the code can be further improved:
> 
> 1. delete 163 lines of code in FlinkSubmitTrait:
>    `.setSavepointRestoreSettings(submitRequest.savepointRestoreSettings)`
> 2. `savepointRestoreSettings` need  improvement
> 
> before:
> 
> ```
>   lazy val savepointRestoreSettings: SavepointRestoreSettings = {
>     lazy val allowNonRestoredState = Try(extraParameter.get(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE.key).toString.toBoolean).getOrElse(false)
>     savePoint match {
>       case sp if Try(sp.isEmpty).getOrElse(true) => SavepointRestoreSettings.none
>       case sp => SavepointRestoreSettings.forPath(sp, allowNonRestoredState)
>     }
>   }
> ```
> 
> after:
> 
> ```
>   lazy val allowNonRestoredState = Try(extraParameter.get(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE.key).toString.toBoolean).getOrElse(false)
> ```
> 
> 3. add code line 101 in FlinkSubmitTrait:
> 
> ```
> .safeSet(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE, submitRequest.allowNonRestoredState)
> ```

Thanks for your review, addressed.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**